### ISSUE TITLE
leaving user in state of uncertainty on changes made

### ISSUE THREAD

**Issue by beyagit** [microsoft/vscode-copilot-release#3138]
Type: <b>Feature Request</b>

in copilot edits, working set is evaluated for user given task and GitHub copilot prompts user to 'Accept' but user does not see any changes to files marked. It is obvious nothing substantial was done because it did not find anything to change. If no changes are done to working set or any other files it should not present accept or discard button, but rather tell the user no changes were found and made. It would be better if in such cases if copilot could suggest alternative ways to approach the issue user is trying to resolve.

Extension version: 0.22.4
VS Code version: Code 1.95.3 (f1a4fb101478ce6ec82fe9627c43efbf9e98c813, 2024-11-13T14:50:04.152Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Remote OS version: Linux x64 6.6.63-0-lts
Remote OS version: Linux x64 6.6.63-0-lts

**Comment by joyceerhl** [comment]
Can you please provide reproducible steps for your scenario? Screenshot or recording would also help, thanks!

**Comment by beyagit** [comment]
well i don't know how to reproduce it. it was happening very often for me but after next update it did not occur at all until today. as soon as it occurred i have made snapshots. i hope this gives a visual of my verbal explanation.
![Image](https://github.com/user-attachments/assets/9d9325f6-c587-4834-aa76-c970cfba1c61)
note that it thinks it made changes in the working set in the edits. but it did not make any change. it is easy for me to know this because working set is small and also now it shows number of lines removed and added (which also did not occur).

here is the result of letting it know no change was made;
![Image](https://github.com/user-attachments/assets/a3720b4d-f68c-41f7-bb85-64c482dd7fe2)
it repeats itself and again brings no change and its response says it made changes. I think it is obvious that it did not find any problems on its own in the code to discriminate any part that does not fit the objectives. I was saying when this occurs, it need to go a step further to re-think the task and why user need might not be achieved and give areas to check for user. this might include confirming user objectives also.

**Comment by joyceerhl** [comment]
It looks like the model did propose changes but we didn't apply them to the file. @aeschli any ideas?

@beyagit could you possibly share the files where this occurs and the prompt you used with Copilot Edits?

**Comment by jrieken** [comment]
/duplicate of https://github.com/microsoft/vscode-copilot-release/issues/3076

**Comment by jrieken** [comment]
> It looks like the model did propose changes but we didn't apply them to the file. @aeschli any ideas?

@joyceerhl It might also be edits that actually didn't change anything? The UI would show the same way? I do have changes ready (for after the break) that would at least fix the rendering 

@beyagit Despite this being a duplicate, repro files and/or log from Copilot Chat (F1 > Show Log > Copilot Chat) would be great to have

**Comment by jrieken** [comment]
@beyagit would you mind updating to latest stable `1.96.x`

**Comment by beyagit** [comment]
@jrieken, yes in my case the edits did not change anything in action but verbally it acts as if it has. I have checked this myself and also questioned Edits on what lines were changed; eventually making it admit to not having made any changes.  Not easy to get it to focus on the lines that were changed because it cannot find it; leads it to do something to the same effect as last as correction.

@joyceerhl, kind of difficult to provide the files, as i have moved on from then and not sure can find exact versions. Also, many updates to the extension since makes it pointless trying to get the file versions as the behavior might no longer exist. I think it is best i come back if the situation repeats. Hopefully with what can recreate the situation.

**Comment by beyagit** [comment]
@jrieken, both vs code and the extension auto updates.
That makes me wonder if it is possible that this is happening at the time when extension has updated but not restarted or has restarted also but vs code has not been restarted.

**Comment by beyagit** [comment]
It seems edits does continue to think it made changes even at simpler tasks and show accept button when no changes were made. I am of the opinion this cannot be something happening to me only and there is need for me to address the issue with case example and methods to reproduce. So, it must be inherent issue that worsens and improves based on general changes in and around the extension. So, no point keeping this open. Please close the case at your will.
